File: JsoupWebtoonCrawling/app/src/main/java/com/example/jsoupwebtooncrawling/MainActivity.java
package com.example.jsoupwebtooncrawling;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.AsyncTask;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.io.IOException;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private String[] week = { "mon", "tue", "wed", "thu", "fri", "sat", "sun" };
    private String TargetURL = "https://comic.naver.com/webtoon/weekdayList.nhn?week=";
    private String TAG = "JsoupMain";
    private String htmlContentInStringFormat="";

    ArrayList<Webtoon> webtoons = new ArrayList<>();

    private Button btn1;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btn1 = findViewById(R.id.btn1);
        btn1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                JsoupAsyncTask jsoupAsyncTask = new JsoupAsyncTask();
                jsoupAsyncTask.execute();
            }
        });
    }

    private class JsoupAsyncTask extends AsyncTask<Void, Void, Void> {

        @Override
        protected void onPreExecute() {
            super.onPreExecute();
        }

        @Override
        protected Void doInBackground(Void... params) {
            try {
                int dayOfWeek = 1;
                Document doc = Jsoup.connect(TargetURL + week[dayOfWeek]).get();

                Elements titles= doc.select("ul.img_list li");

                System.out.println("-------------------------------------------------------------");
                for(Element e: titles) {
                    String title = e.select("dl dt a").attr("title");
                    String writer = e.select("dl dd.desc a").text();
                    String star = e.select("dl dd div.rating_type strong").text();
                    String thumbnailSrc = e.select("div.thumb a img").attr("src");
                    String webtoonURL = "https://comic.naver.com" + e.select("div.thumb a").attr("href");

                    webtoons.add(new Webtoon(title, writer, star, thumbnailSrc, webtoonURL));
                    Log.d(TAG, title + ", " + writer + ", " + star + ", " + thumbnailSrc + ", " + webtoonURL);
                }
                Intent intent = new Intent(getApplicationContext(), WebtoonActivity.class);
                intent.putExtra("webtoons", webtoons);
                startActivity(intent);
            } catch (IOException e) {
                Toast.makeText(getApplicationContext(), "정보를 불러오는데 실패했습니다.", Toast.LENGTH_SHORT).show();
                e.printStackTrace();
            }

            return null;
        }

        @Override
        protected void onPostExecute(Void result) {

        }

    }
}

File: AndroidKotlin/ListViewKt2/app/src/main/java/com/example/listviewkt2/MainActivity.kt
package com.example.listviewkt2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

     var UserList = arrayListOf<User>(
         User(R.drawable.batman, "양태웅", "17", "안녕하세요"),
         User(R.drawable.batman, "엄태웅", "17", "안녕하세요"),
         User(R.drawable.batman, "김태웅", "17", "안녕하세요"),
         User(R.drawable.batman, "박태웅", "17", "안녕하세요"),
         User(R.drawable.batman, "이태웅", "17", "안녕하세요"),
         User(R.drawable.batman, "정태웅", "17", "안녕하세요"),
         User(R.drawable.batman, "최태웅", "17", "안녕하세요"),
         User(R.drawable.batman, "성태웅", "17", "안녕하세요")
     )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val item = arrayOf("사과", "배", "자두", "딸기", "키위")

        // context는 한 액티비티의 모든 정보를 담고 있다.
        val Adapter = UserAdapter(this, UserList)
        listView.adapter = Adapter

        listView.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            val selectItem = parent.getItemAtPosition(position) as User
            Toast.makeText(this, selectItem.name, Toast.LENGTH_SHORT).show()

        }
    }
}